TLM/TLM/Util/Record/LaneArrowsRecord.cs

```diff
@@ -31,24 +29,32 @@ public class LaneArrowsRecord : IRecordable {
             LaneArrowManager.Instance.SetLaneArrows(laneId, arrows_.Value);
         }
 
+        /// <summary>
+        /// Obtain lane arrow records for the lanes exiting the specified
+        /// end of a segment.
+        /// </summary>
+        /// <param name="segmentId">The id of the segment to inspect.</param>
+        /// <param name="startNode">
+        /// If <c>true</c>, get lanes exiting via the start node; otherwise
+        /// get lanes exiting via the end node.</param>
+        /// <returns>
+        /// Returns a sorted list of <see cref="LaneArrowsRecord"/> associated
+        /// with each lane exiting via the specified segment end. The list may
+        /// be empty if no matching lanes were found.
+        /// </returns>
         public static List<LaneArrowsRecord> GetLanes(ushort segmentId, bool startNode) {
-            int maxLaneCount = segmentId.ToSegment().Info.m_lanes.Length;
-            var ret = new List<LaneArrowsRecord>(maxLaneCount);
-            ExtSegmentManager extSegmentManager = ExtSegmentManager.Instance;
-            var lanes = extSegmentManager.GetSortedLanes(
-                segmentId,
-                ref segmentId.ToSegment(),
+
+            var lanes = segmentId.ToSegment().GetSortedLanes(
                 startNode,
                 LaneArrowManager.LANE_TYPES,
                 LaneArrowManager.VEHICLE_TYPES,
                 sort: false);
-            foreach (var lane in lanes) {
-                LaneArrowsRecord laneData = new LaneArrowsRecord {
-                    LaneId = lane.laneId,
-                };
-                ret.Add(laneData);
-            }
-            ret.TrimExcess();
+
+            var ret = new List<LaneArrowsRecord>(lanes.Count);
```

Comment:
Can this be zero size? Return a premade static zero size readonly list in such case, do not allocate.

Reply:
it does need the space.